src/vmm/src/device_manager/mmio.rs

```diff
@@ -285,19 +285,14 @@ impl MMIODeviceManager {
     /// otherwise allocate a new MMIO slot for it.
     pub fn register_mmio_rtc(
         &mut self,
-        vm: &VmFd,
         rtc: Arc<Mutex<devices::legacy::RTC>>,
         dev_info_opt: Option<MMIODeviceInfo>,
     ) -> Result<()> {
         // Create and attach a new RTC device.
+        // We allocate an IRQ even though we do not need it so that
+        // we do not break snapshot compatibility.
```

Comment:
I think we need to update changelog to mention the v0.24 RTC issue was fixed, how RTC works now and that we allocate a dummy IRQ starting with v0.25. Customers might wonder why there is a gap in allocated IRQs.

Reply:
Do we need to keep allocating the dummy IRQ in v0.25 from-scratch-booted VMs?

Or are you only referring to microVMs loaded from snapshots generated with < v0.25 Firecracker versions?

Reply:
I don't think we break compatibility if we remove the dummy IRQ.

Reply:
Yes, I was referring to microVms loaded from snapshots generated with v0.24. I will test it out.

Reply:
v0.24 -> v0.25 should work fine, but the dummy IRQ is for v0.25 -> v0.24 which will actually register that irqfd.

Reply:
You're both right, resolving this.